### PR TITLE
specify 'dotnet-ef' version for tools-in-path

### PR DESCRIPTION
**Pull request by nicrowe00**
With the release of .NET 8, the tools for .NET have also been updated for .NET 8. The new updated tools are incompatiable with .NET 6 and 7 meaning the correct tool version must be specified for the version of .NET installed on the machine.

### CODE REVIEW

**Comment by nicrowe00** [comment]
The test for .NET 8 on Fedora 38 is failing as the new release of .NET 8 still isn't available in the package manager for Fedora which causes it to use .NET 8-rc2 which is incompatible with the new update for 'dotnet-ef'. This test will continue to fail until new release of .NET 8 is added to the Fedora package manager.

**Comment by nicrowe00** [comment]
@tmds @omajid PTAL

**Comment by tmds** [comment]
Can you try what happens if we replace the whole `if .. fi` block with:
```sh
IFS='.-' read -ra VERSION_SPLIT <<< "$1" 

dotnet tool update -g dotnet-ef --version "${VERSION_SPLIT[0]}.*-*"
```

**Comment by tmds** [comment]
> The new updated tools are incompatiable with .NET 6 and 7

For reference: https://github.com/dotnet/sdk/issues/36904

File: tools-in-path/test.sh [hunk 1]
```diff
@@ -19,10 +19,13 @@ else
     exit 1
 fi
 
+version=$(dotnet --version)
```
Comment (tmds) [inline tools-in-path/test.sh:22]: Can you replace everything including this version variable up to the `fi` with these two lines:
```
IFS='.-' read -ra VERSION_SPLIT <<< "$version"
dotnet tool update -g dotnet-ef --version "${VERSION_SPLIT[0]}.*-*"
```

If the `|| true` is needed, it would be nice if we add a comment that explains why.

Comment (tmds) [reply tools-in-path/test.sh:22]: Note: `$1` is set by the test runner to the runtime version.

Comment (nicrowe00) [reply tools-in-path/test.sh]: > Note: `$1` is set by the test runner to the runtime version.

Thank you, I wasn't aware of this.

**Review by omajid — APPROVED** [review]
Good catch!

File: tools-in-path/test.sh [hunk 2]
```diff
@@ -7,7 +7,7 @@ if [ -f /etc/profile ]; then
 fi
 
 set -euo pipefail
-
+IFS=$'\n\t'
```
Comment (tmds) [inline tools-in-path/test.sh:10]: Why did you add this?

Comment (omajid) [reply tools-in-path/test.sh:10]: I assume because it's generally added next to `set -euo pipefail` as part of ["bash strict mode"](http://redsymbol.net/articles/unofficial-bash-strict-mode/)?

Comment (tmds) [reply tools-in-path/test.sh:10]: Should we be adding it to all tests then?

Comment (omajid) [reply tools-in-path/test.sh:10]: I would be okay with that. Doesn't seem very high priority though.
